### PR TITLE
Fix: Reinforce the WatchTower implementation 

### LINKED ISSUE #121: Reinforce the WatchTower implementation
## Description 📹

**Replace println & eprintln by WatchTower (our own logger) as done in src/main.rs**
It looks like that:
```rs
    logger.info(format!(
        "id_server: starting v{}",
        env!("CARGO_PKG_VERSION")
    ));
```
(Don't forget to also use logger.warning & logger.severe when appropriate)

## Proposed Actions 🛠️

Here’s a checklist of actions to follow for resolving this issue:

1. **Fork and Create Branch**:  
   Fork the repository and create a new branch using the issue number:
   ```bash
   git checkout -b fix-[issue-number]
   ```

2. **Implement Changes**:  
- [ ] Use the global search tool provided by your IDE to find all occurences of println and replace them, if possible, by our logger

3. **Run Tests and Commit Changes**:  
   Make sure your changes don't break existing functionality and commit with a clear message:
   ```bash
   git commit -m "Fix: [Short description of the fix]"
   ```

## Required 📋

To keep our workflow smooth, please make sure you follow these guidelines:

- **Assignment**: Don't create a pull request if you weren’t assigned to this issue.
- **Timeframe**: Complete the task within **3 business days**.
- **Closing the Issue**: In your PR description, close the issue by writing `Close #[issue_id]`.
- **Review Process**:
  - Once you've submitted your PR, change the label to **"ready for review"**.
  - If changes are requested, address them and then update the label back to **"ready for review"** once done.
- **Testing**: Test your PR locally before pushing, and verify that tests and build are working after pushing.

Thank you for your contribution 🙏

⚠️ WARNING: Failure to follow the requirements above may result in being added to the OnlyDust blacklist, affecting your ability to receive future rewards.

### PR DESCRIPTION
**Pull request by ryzen-xp**
close: #121

### CODE REVIEW

**Comment by ryzen-xp** [comment]
@Marchand-Nicolas  Sir Please review th PR

File: src/endpoints/campaigns/get_free_domain.rs [hunk 1]
```diff
@@ -35,6 +35,8 @@ pub async fn handler(
     State(state): State<Arc<AppState>>,
     Query(query): Query<FreeDomainQuery>,
 ) -> impl IntoResponse {
+    let conf = config::load();
+    let logger = Logger::new(&conf.watchtower);
```
Comment (Marchand-Nicolas) [inline src/endpoints/campaigns/get_free_domain.rs]: We don't want to have multiple logger instance. You can use &state.logger instead (which returns the logger instance created in main.rs)

**Review by Marchand-Nicolas — CHANGES_REQUESTED** [review]

File: src/main.rs [hunk 2]
```diff
@@ -31,7 +31,7 @@ mod tests;
 
 #[tokio::main]
 async fn main() {
-    println!("starknetid_server: starting v{}", env!("CARGO_PKG_VERSION"));
+    // println!("starknetid_server: starting v{}", env!("CARGO_PKG_VERSION"));
```
Comment (Marchand-Nicolas) [inline src/main.rs]: Don't comment this line. Instead, move it after the logger definition, and use it

**Review by Marchand-Nicolas — CHANGES_REQUESTED** [review]

**Comment by ryzen-xp** [comment]
@Marchand-Nicolas  Sir  please check  again my PR  ,

File: src/resolving.rs [hunk 3]
```diff
@@ -36,9 +36,8 @@ pub async fn get_custom_resolver(domains: &Collection<Document>, domain: &str) -
         Ok(doc) => {
             if let Some(document) = doc {
                 // If the resolver field exists, is not null, and is not 0x000... then return it
-                if let Some(resolver) = document.get_str("resolver").ok() {
-                    if resolver
-                        != "0x0000000000000000000000000000000000000000000000000000000000000000"
+                if let Ok(resolver) = document.get_str("resolver") {
```
Comment (Marchand-Nicolas) [inline src/resolving.rs]: Why this change ?

**Review by Marchand-Nicolas — CHANGES_REQUESTED** [review]

File: src/resolving.rs (hunk 3 above)
Comment (ryzen-xp) [reply src/resolving.rs]: sorry sir I did not notice

**Comment by ryzen-xp** [comment]
@Marchand-Nicolas Sir please check again my PR .

**Review by Marchand-Nicolas — APPROVED** [review]
Lgtm!

**Comment by ryzen-xp** [comment]
Thank You Sir @Marchand-Nicolas

**Pull request merged**